File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        if (nums.empty()) return 0;

        int count=1;
        int maxVal = 1;

        sort(nums.begin(),nums.end());

        for(int i=1;i<nums.size();i++){
            if(nums[i]==nums[i-1]){
                continue;
            }
            else if(nums[i]==nums[i-1]+1){
                count++;
            }
            else{
                maxVal = max(maxVal,count);
                count=1;
            }
        }
        return max(maxVal,count);
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set <Integer> mySet= new HashSet<>();

        for(int num:nums){
            mySet.add(num);
        }

        int longestStreak=0;

        for(int num:mySet){
            if(!mySet.contains(num-1)){
                int currentNum= num;
                int currentStreak=1;

                  while(mySet.contains(currentNum+1)){
                currentNum++;
                currentStreak++;
            }
             longestStreak=Math.max(longestStreak, currentStreak);
            }
          
           
        }
        return longestStreak;
    }
}